pkg/scheduler/initialsizeclass/platform_overriding_size_class_analyzer.go

```diff
@@ -0,0 +1,91 @@
+package initialsizeclass
+
+import (
+	"context"
+	"strconv"
+	"time"
+
+	remoteexecution "github.com/bazelbuild/remote-apis/build/bazel/remote/execution/v2"
+	"github.com/buildbarn/bb-storage/pkg/digest"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+type platformOverridingSizeClassAnalyzer struct {
+	platformKey            string
+	actionTimeoutExtractor *ActionTimeoutExtractor
+	base                   Analyzer
+}
+
+func NewPlatformOverridingAnalyzer(actionTimeoutExtractor *ActionTimeoutExtractor, platformKey string, base Analyzer) Analyzer {
+	return &platformOverridingSizeClassAnalyzer{
+		platformKey:            platformKey,
+		actionTimeoutExtractor: actionTimeoutExtractor,
+		base:                   base,
+	}
+}
+
+type sizeClassSelector struct {
+	sizeClass string
+	timeout   time.Duration
+}
+
+func (a *platformOverridingSizeClassAnalyzer) Analyze(ctx context.Context, digestFunction digest.Function, action *remoteexecution.Action) (Selector, error) {
+	timeout, err := a.actionTimeoutExtractor.ExtractTimeout(action)
+	if err != nil {
+		return nil, err
+	}
+
+	platform := action.Platform
+	for _, key := range platform.Properties {
```

Comment:
Should this also use platform.KeyExtractor, instead of trying to read the Platform message from Action directly? It could also be stored in the Command message.